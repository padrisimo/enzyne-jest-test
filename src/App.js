import React, { Component } from 'react'
import logo from './logo.svg'
import './App.css'

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">Hello World</p>
      </div>
    )
  }
}

export class Link extends Component {
  render() {
    return this.props.hide ? null : <a href={this.props.address}>Click</a>
  }
}

export default App