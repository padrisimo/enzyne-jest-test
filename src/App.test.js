import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import App from './App';

configure({ adapter: new Adapter() });

describe('<App />', () => {
  it('should contain 1 p element and h1 text', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find('p').length).toBe(1)
    expect(wrapper.find('h1').text()).toBe('Welcome to React')
  })
})  
