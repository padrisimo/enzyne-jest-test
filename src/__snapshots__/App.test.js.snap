// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> mount rendering matches the snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <h1
        className="App-title"
      >
        Welcome to React
      </h1>
    </header>
    <p
      className="App-intro"
    >
      Hello World
    </p>
  </div>
</App>
`;

exports[`<App /> shallow rendering matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    Hello World
  </p>
</div>
`;
